packages/bot/src/types/CallbackTypes.ts

```diff
@@ -1,4 +1,4 @@
 import { Client } from '../Client';
 import { Message } from '../models/Message';
 
-export type MessageHandler<T = void> = (msg: Message, client: Client) => T;
+export type MessageHandler<T = void, U extends Client = Client> = (msg: Message, client: U) => T;
```

Comment:
Does the generic T make any sense?

Reply:
While I was making the change, I noticed that there was, indeed, a case where having a generic type parameter for the return type of the function signature made sense - some of the `MessageHandler`s return `boolean` and are used as validators. However, I didn't find any other return types, so simply settled on `boolean | void | Promise<void>` - that was enough to cover all cases. 

Had to apply a small fix for `RegisterShutdownScript`, though, as it used a concise body and thus ended up returning `Promise<Promise<void>[]>`. On a related note, there was a strange thing about it: you had `Object.values()` and a `// @ts-ignore` comment whereas `Object.entries()` was supposed to be used judging from the context. I fixed it, but not sure if there was a very specific reason in doing so.